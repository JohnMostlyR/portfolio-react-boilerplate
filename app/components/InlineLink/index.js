import styled from 'styled-components';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { theme } from '../../styles/theme';

const { color, textShadowColor } = theme.inlineLink;

const InlineLink = styled(Link)`
  background-color: transparent;
  background-image: linear-gradient(currentColor, currentColor);
  background-origin: padding-box;
  background-position: center bottom 10%;
  background-repeat: no-repeat;
  background-size: 100% 1.5px;
  color: ${color};
  cursor: pointer;
  text-decoration: none;
  text-shadow: 3px 0 ${textShadowColor}, 2px 0 ${textShadowColor},
    1px 0 ${textShadowColor}, -1px 0 ${textShadowColor},
    -2px 0 ${textShadowColor}, -3px 0 ${textShadowColor};
  touch-action: manipulation;
  transition: color 0.1s ease-out;

  @supports (text-decoration-skip-ink: auto) {
    background: none;
    text-decoration: underline;
    text-decoration-skip-ink: auto; /* stylelint-disable-line property-no-unknown */
    text-shadow: none;
  }
`;

InlineLink.propTypes = {
  to: PropTypes.string.isRequired,
};

export default InlineLink;
