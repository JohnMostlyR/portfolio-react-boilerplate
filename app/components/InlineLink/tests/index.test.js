import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import InlineLink from '../index';

describe('<InlineLink />', () => {
  it('Should render and match the snapshot', () => {
    const renderedComponent = shallow(<InlineLink to="/" />);
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });
});
