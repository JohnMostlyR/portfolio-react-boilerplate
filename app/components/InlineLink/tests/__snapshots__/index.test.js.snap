// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineLink /> Should render and match the snapshot 1`] = `
.c0 {
  background-color: transparent;
  background-image: linear-gradient(currentColor,currentColor);
  background-origin: padding-box;
  background-position: center bottom 10%;
  background-repeat: no-repeat;
  background-size: 100% 1.5px;
  color: #00599e;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: 3px 0 #fff,2px 0 #fff,1px 0 #fff,-1px 0 #fff,-2px 0 #fff,-3px 0 #fff;
  touch-action: manipulation;
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
}

@supports (-webkit-text-decoration-skip-ink:auto) or (text-decoration-skip-ink:auto) {
  .c0 {
    background: none;
    -webkit-text-decoration: underline;
    text-decoration: underline;
    -webkit-text-decoration-skip-ink: auto;
    text-decoration-skip-ink: auto;
    text-shadow: none;
  }
}

<Link
  className="c0"
  replace={false}
  to="/"
/>
`;
